src/client/eppo-client.spec.ts

```diff
@@ -217,34 +223,48 @@ describe('EppoClient E2E test', () => {
   });
 
   it('returns subject from overrides when enabled is true', () => {
-    window.localStorage.setItem(
-      flagKey,
-      JSON.stringify({
-        ...mockExperimentConfig,
-        typedOverrides: {
-          '1b50f33aef8f681a13f623963da967ed': 'control',
-        },
-      }),
-    );
+    const entry = {
+      ...mockExperimentConfig,
+      enabled: false,
+      overrides: {
+        '1b50f33aef8f681a13f623963da967ed': 'override',
+      },
+      typedOverrides: {
+        '1b50f33aef8f681a13f623963da967ed': 'override',
+      },
+    };
+
+    storage.setEntries({ [flagKey]: entry });
+
     const client = new EppoClient(storage);
+    const mockLogger = td.object<IAssignmentLogger>();
+    client.setLogger(mockLogger);
+
     const assignment = client.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual('control');
+    expect(assignment).toEqual('override');
+    expect(td.explain(mockLogger.logAssignment).callCount).toEqual(0);
   });
 
   it('returns subject from overrides when enabled is false', () => {
     const entry = {
       ...mockExperimentConfig,
       enabled: false,
+      overrides: {
+        '1b50f33aef8f681a13f623963da967ed': 'override',
+      },
       typedOverrides: {
-        '1b50f33aef8f681a13f623963da967ed': 'control',
+        '1b50f33aef8f681a13f623963da967ed': 'override',
       },
     };
 
     storage.setEntries({ [flagKey]: entry });
 
     const client = new EppoClient(storage);
+    const mockLogger = td.object<IAssignmentLogger>();
+    client.setLogger(mockLogger);
     const assignment = client.getAssignment('subject-10', flagKey);
-    expect(assignment).toEqual('control');
```

Comment:
The variation value is "control" when we don't override too, so this is a bit unhelpful

Reply:
good catch and nice change!